crates/ruff_python_formatter/resources/test/fixtures/ruff/statement/with.py

```diff
@@ -0,0 +1,52 @@
+with aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa, aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa:
+    ...
+    # trailing
+
+with a, a:  # after colon
+    ...
+    # trailing
+
+with (
+    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+    aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+):
+    ...
+    # trailing
+
+
+with call(arg):
+    call(arg)
+# fmt: skip
+
+
+with (
+        a  # a
+        ,  # comma
+        b  # c
+        ): # colon
+    ...
+
+
+with (
+        a  # a
+        as  # as
+        b  # b
+        ,  # comma
+        c  # c
+        ): # colon
+    ...  # body
+    # body trailing own
+
+
+with (a,):  # magic trailing comma
+    ...
+
+
+with (a):  # should remove brackets
+    ...
+
+# TODO: black doesn't wrap this, but maybe we want to anyway?
```

Comment:
This sounds reasonable to me. We should make it wrap the same as for function definition

```python
# All flat
def test(a, b, c): 
	pass

# parentheses expanded
def test(
	a, b, c
): 
	pass

# all expanded
def test(
	a, 
	b, 
	c
): 
	pass
```

You can achieve this by using two nested groups:
* outer group: For the parentheses and indent
* inner group: Around the with items

Reply:
oh wait. when there is an `as`, wrapping isn't allowed in python < 3.9 (3.8 isn't eol until oct '24)

Reply:
We can change that when adding support for targeting different python versions. It's something that we'll need to add to the formatter options.